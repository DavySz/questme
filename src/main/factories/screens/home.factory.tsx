import { HomeContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeHomeScreen(): ReactNode {
  return <HomeContainer />;
}
