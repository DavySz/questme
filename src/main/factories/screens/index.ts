export * from "./onboarding-factory";
export * from "./splash-factory";
export * from "./initial-factory";
export * from "./login-factory";
export * from "./sign-up-factory";
