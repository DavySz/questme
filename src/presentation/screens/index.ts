export * from "./home/home.container";
export * from "./splash/splash.container";
export * from "./onboarding/onboarding.container";
export * from "./initial/initial.container";
export * from "./login/login.container";
export * from "./sign-up/sign-up.container";
export * from "./reset-password/reset-password.container";
export * from "./new-password/new-password.container";
export * from "./sign-up-email";
