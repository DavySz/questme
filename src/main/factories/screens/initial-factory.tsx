import { InitialContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeInitialScreen(): ReactNode {
  return <InitialContainer />;
}
