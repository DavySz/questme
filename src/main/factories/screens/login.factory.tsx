import { LoginContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeLoginScreen(): ReactNode {
  return <LoginContainer />;
}
