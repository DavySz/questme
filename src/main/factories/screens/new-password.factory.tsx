import { NewPasswordContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeNewPasswordScreen(): ReactNode {
  return <NewPasswordContainer />;
}
