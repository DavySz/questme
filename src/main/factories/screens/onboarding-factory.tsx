import { OnboardingContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeOnboardingScreen(): ReactNode {
  return <OnboardingContainer />;
}
