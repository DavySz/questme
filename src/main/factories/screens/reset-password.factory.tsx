import { ResetPasswordContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeResetPasswordScreen(): ReactNode {
  return <ResetPasswordContainer />;
}
