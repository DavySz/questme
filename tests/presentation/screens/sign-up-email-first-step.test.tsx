import { SignUpEmailFirstStepContainer } from "@presentation/screens";
import { fireEvent, render } from "@testing-library/react-native";
import { TestingPublicNavigator, customRender } from "../../utils";

describe("SignUpEmailFirstStep", () => {
  test("Should render screen without errors", () => {
    const tree = customRender(<SignUpEmailFirstStepContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Should navigate to correct screen when go back button is pressed", async () => {
    const { getByTestId, findByTestId } = render(
      <TestingPublicNavigator currentRoute="sign-up" />
    );

    const buttonSignUpWithEmail = getByTestId("button-sign-up-with-email");
    fireEvent.press(buttonSignUpWithEmail);

    const goBackButton = getByTestId("go-back-button");
    fireEvent.press(goBackButton);

    const previousScreen = await findByTestId("sign-up-screen-content");

    expect(previousScreen).toBeTruthy();
  });

  test("Should navigate to second step screen when continue button is clicked", async () => {
    const { getByText, findByTestId } = render(
      <TestingPublicNavigator currentRoute="sign-up-email-first-step" />
    );

    const buttonSecondStep = getByText("Continue");
    fireEvent.press(buttonSecondStep);

    const secondStepScreen = await findByTestId(
      "sign-up-email-second-step-screen-content"
    );

    expect(secondStepScreen).toBeTruthy();
  });
});
