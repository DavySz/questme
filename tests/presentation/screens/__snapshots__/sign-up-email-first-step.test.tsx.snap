// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpEmailFirstStep Should render screen without errors 1`] = `
<View
  style={
    {
      "backgroundColor": "#EFEEFC",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
  testID="default-container"
  variant="secondary"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "left": 16,
          "position": "absolute",
          "top": 8,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomLeftRadius": 20,
            "borderBottomRightRadius": 20,
            "borderTopLeftRadius": 20,
            "borderTopRightRadius": 20,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
        testID="go-back-button"
      >
        <View
          side="none"
          style={{}}
          testID="icon-wrapper"
        >
          <SvgMock
            color="#0C092A"
            testID="icon-with-color"
          />
        </View>
      </View>
    </View>
    <Text
      color="neutral-black"
      style={
        {
          "color": "#0C092A",
          "fontFamily": "Rubik_500Medium",
          "fontSize": 24,
          "lineHeight": 36,
        }
      }
      testID="header-title"
      variant="heading-3"
    >
      Whatâ€™s your email?
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "flex": 1,
        "padding": 24,
      }
    }
    showsVerticalScrollIndicator={false}
    style={{}}
    testID="sign-up-email-first-step-screen-content"
    variant="secondary"
  >
    <View>
      <View
        style={
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <View
          style={
            {
              "marginBottom": 8,
            }
          }
        >
          <Text
            color="neutral-black"
            style={
              {
                "color": "#0C092A",
                "fontFamily": "Rubik_400Regular",
                "fontSize": 14,
                "lineHeight": 20,
              }
            }
            variant="body-small-regular"
          >
            Email Address
          </Text>
        </View>
        <View
          isFocused={false}
          style={
            {
              "backgroundColor": "#FFF",
              "borderBottomLeftRadius": 20,
              "borderBottomRightRadius": 20,
              "borderColor": "#EFEEFC",
              "borderStyle": "solid",
              "borderTopLeftRadius": 20,
              "borderTopRightRadius": 20,
              "borderWidth": 2,
              "flexDirection": "row",
              "gap": 16,
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": "100%",
            }
          }
          testID="input-wrapper"
        >
          <SvgMock
            color="#6A5AE0"
            testID="icon-with-color"
          />
          <TextInput
            onBlur={[Function]}
            onFocus={[Function]}
            placeholder="Your email address"
            style={
              {
                "fontFamily": "Rubik_400Regular",
                "fontSize": 16,
                "width": "100%",
              }
            }
            testID="custom-input"
          />
        </View>
      </View>
      <View
        style={
          {
            "gap": 24,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "flex-end",
              "gap": 8,
              "width": "100%",
            }
          }
        >
          <Text
            color="primary"
            style={
              {
                "color": "#6A5AE0",
                "fontFamily": "Rubik_500Medium",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
            variant="body-normal-medium"
          >
            1
             of 
            3
          </Text>
          <View
            style={
              {
                "width": "100%",
              }
            }
          >
            <View
              progressPercentage={33.33333333333333}
              style={
                {
                  "backgroundColor": "#6A5AE0",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "height": 8,
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": "33.33333333333333%",
                }
              }
            />
            <View
              style={
                {
                  "backgroundColor": "#6A5AE0",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "height": 8,
                  "left": 0,
                  "opacity": 0.1,
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#6A5AE0",
              "borderBottomLeftRadius": 20,
              "borderBottomRightRadius": 20,
              "borderTopLeftRadius": 20,
              "borderTopRightRadius": 20,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": "100%",
            }
          }
        >
          <Text
            color="neutral-white"
            style={
              {
                "color": "#FFF",
                "fontFamily": "Rubik_500Medium",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
            variant="body-normal-medium"
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
